pkg/globalnet/controllers/ipam/gatewaymonitor.go

```diff
@@ -77,6 +77,7 @@ func NewGatewayMonitor(spec *SubmarinerIpamControllerSpecification, cfg *rest.Co
 
 func (i *GatewayMonitor) Run(stopCh <-chan struct{}) error {
 	defer utilruntime.HandleCrash()
+	defer ClearGlobalNetChains(i.ipt)
```

Comment:
I wonder if we may add it right after L95.

I'm not sure if HandleCrash could fire before and stop the process? (not sure how it works)

Just a question.

Reply:
Thanks for reviewing @mangelajo 

When multiple defer() calls are used, they are executed in Last-in-first-out order.
Also, I did some stress/monkey-testing with a script that was removing the label from the active gateway node and configuring it on a new node in a while loop with a small delay in between. I found that using defer was more reliable (aka doing the necessary cleanup) than including it at L95. Hence moved the cleanup to L80 and used defer.